compiler/src/dmd/constfold.d

```diff
@@ -1271,20 +1271,59 @@ UnionExp Slice(Type type, Expression e1, Expression lwr, Expression upr)
         const uinteger_t ilwr = lwr.toInteger();
         const uinteger_t iupr = upr.toInteger();
         if (sliceBoundsCheck(0, es1.elements.length, ilwr, iupr))
-            cantExp(ue);
+        {
+            bool isError = false;
+            if (lwr.op == EXP.int64 && upr.op == EXP.int64 && ilwr > iupr)
+            {
+                e1.error("in slice `%s[%llu..%llu]`, lower bound is greater than upper bound", e1.toChars, ilwr, iupr);
```

Comment:
Deduplicate these. 

There seems to be a pattern in the rightmost subtree of the if conditions that could be normalized too

Reply:
Deduplicated the `.op` checks; dunno what other pattern you ment ^^'